lib/cryptopunk/crypto/tron/validation.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Cryptopunk.Crypto.Tron.Validation do
+  @moduledoc false
+
+  alias Cryptopunk.Crypto.Tron.ChecksumEncoding
+
+  @mixed_case_regexp ~r/^T[A-Za-z1-9]{33}$/i
+  @upcase_regexp ~r/^T[1-9A-Z]{33}$/
+  @downcase_regexp ~r/^T[1-9a-f]{33}$/
+
+  @spec valid?(String.t()) :: boolean()
+  def valid?(address) do
+    with true <- address_valid_format(address),
```

Comment:
could be done with ifs

Reply:
@pxp9
[rework checksum encoding to avoid unnecessary with expr and allow imp…](https://github.com/ayrat555/cryptopunk/pull/74/commits/f97524631a62344658b7c2ac80a7ba71d5422423)